subsys/ipc/ipc_service/backends/ipc_icmsg.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/mbox.h>
+
+#include <ipc/ipc_service_backend.h>
+#include "ipc_icmsg.h"
+
+#define DT_DRV_COMPAT	zephyr_ipc_icmsg
+#define CB_BUF_SIZE	CONFIG_IPC_SERVICE_BACKEND_ICMSG_CB_BUF_SIZE
+
+static const uint8_t magic[4] = {0xDE, 0xAD, 0xBE, 0xEF};
+BUILD_ASSERT(sizeof(magic) <= UINT8_MAX);
+BUILD_ASSERT(CB_BUF_SIZE <= UINT8_MAX);
+
```

Comment:
nit: empty line

Reply:
I will move double empty line between everything thats static (including includes) and actual code. That's similar in `ipc_icmsg_buf.c` file. Im used to it, ii like it this way.